dev/com.ibm.ws.session/src/com/ibm/ws/session/SessionCookieConfigImpl.java

```diff
@@ -203,23 +253,26 @@ public void setSecure(boolean b, boolean externalCall) {
         if (contextInitialized) {
             throwWarning();
         }
-        secure = b;
+        putAttribute(SECURE, String.valueOf(b));
         secureSet = true;
     }
 
     @Override
-    public SessionCookieConfigImpl clone() throws CloneNotSupportedException {
-        SessionCookieConfigImpl temp = new SessionCookieConfigImpl(name, domain, path, comment, maxAge, httpOnly, secure);
+    public SessionCookieConfig clone() throws CloneNotSupportedException {
+        SessionCookieConfigImpl temp = new SessionCookieConfigImpl(getName(), getDomain(), getPath(), comment, getMaxAge(), isHttpOnly(), isSecure());
+        temp.attributes = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
+        temp.attributes.putAll(this.attributes);
+
         return temp;
     }
 
-    private void throwWarning() {
+    protected void throwWarning() {
         String msg = nls.getString("programmatic.sessions.already.been.initialized");
```

Comment:
opened: https://github.com/OpenLiberty/open-liberty/issues/22433 to update the nls here and get a translated message.